apps/src/dance/ai/utils.ts

```diff
@@ -12,18 +13,50 @@ export const generateBlocksFromResult = (
   const blocksSvg = generateBlocks(workspace);
 
   // Foreground block.
-  blocksSvg[0].setFieldValue(effect.foregroundEffect, 'EFFECT');
+  validateAndSetFieldValue(
+    blocksSvg[0].getField('EFFECT') as FieldDropdown,
+    effect.foregroundEffect,
+    FieldKey.FOREGROUND_EFFECT
+  );
 
   // Background block.
-  blocksSvg[1].setFieldValue(effect.backgroundEffect, 'EFFECT');
-  blocksSvg[1].setFieldValue(effect.backgroundColor, 'PALETTE');
+  validateAndSetFieldValue(
+    blocksSvg[1].getField('EFFECT') as FieldDropdown,
+    effect.backgroundEffect,
+    FieldKey.BACKGROUND_EFFECT
+  );
+  validateAndSetFieldValue(
+    blocksSvg[1].getField('PALETTE') as FieldDropdown,
+    effect.backgroundColor,
+    FieldKey.BACKGROUND_PALETTE
+  );
 
   // Connect the blocks.
   blocksSvg[0].nextConnection.connect(blocksSvg[1].previousConnection);
 
   return blocksSvg;
 };
 
+function validateAndSetFieldValue(
+  dropdown: FieldDropdown,
+  value: string,
+  field: FieldKey
+) {
+  if (
+    dropdown
+      .getOptions()
+      .some(option => option[1] === value || option[1] === `"${value}"`)
+  ) {
+    dropdown.setValue(value);
+  } else {
+    Lab2MetricsReporter.logWarning({
```

Comment:
Note: thanks to Jim's PR: https://github.com/code-dot-org/code-dot-org/pull/54913, the reporter should have all the common values (channel ID, app name) already set by the time we hit this